vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -142,7 +145,8 @@ void AdsImpl::InitializeStep4(const Result result) {
   ads_serve_->DownloadCatalog();
 }
 
-void AdsImpl::CleanPostedAsRegistryAfterReboot() {
+#if defined(OS_ANDROID)
```

Comment:
See above comment

Reply:
using preprocessor

Reply:
See below as more platform specific code is being added to the ads library and should be in the service layer